### PR TITLE
Make group context optional for MLSPlaintextTBS

### PR DESCRIPTION
**Pull request by raphaelrobert**
This PR makes the group context field optional for `MLSPlaintextTBS`. 

Note that currently only messages from internal members are supported. A corresponding issue has been filed in #106.

### CODE REVIEW

**Review by kkohbrok — DISMISSED** [review]

File: src/framing/mod.rs [hunk 1]
```diff
@@ -66,7 +66,8 @@ impl MLSPlaintext {
             content,
             signature: Signature::new_empty(),
         };
-        mls_plaintext.sign(ciphersuite, signature_key, context);
+        let serialized_context = context.encode_detached().unwrap();
```
Comment (franziskuskiefer) [inline src/framing/mod.rs:69]: Shouldn't the error get propagated here instead of panicking? The `context` is input and might not be well-formed.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Generally looks good but I'm missing tests.

File: src/framing/mod.rs (hunk 1 above)
Comment (raphaelrobert) [reply src/framing/mod.rs:69]: The codec code throws an error when slices are too long to be encoded with a certain length header (e.g. a slice of 257 elements encoded with a `u8` header).
If we ant to make sure the group context doesn't have this problem, we should check it when constructing it, not when serializing it.
Sanitizing input is something we should improve anyway, so that we don't have to worry about it internally. I filed #115 for that.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Generally looks good but I'm missing tests.

File: src/framing/test_framing.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    let enc = orig.encode_detached().unwrap();
+    let copy = MLSPlaintext::from_bytes(&enc).unwrap();
+    assert_eq!(orig, copy);
+}
+
+#[test]
+fn context_presence() {
+    use crate::ciphersuite::*;
+
+    let ciphersuite =
```
Comment (franziskuskiefer) [inline src/framing/test_framing.rs:58]: Not for this PR but we should write helpers for this. We need this first couple lines in most tests.

File: src/framing/test_framing.rs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        signature: Signature::new_empty(),
+    };
+    let context = GroupContext {
+        group_id: GroupId::random(),
+        epoch: GroupEpoch(1u64),
+        tree_hash: vec![],
+        confirmed_transcript_hash: vec![],
+    };
+    let serialized_context = context.encode_detached().unwrap();
+    let signature_input = MLSPlaintextTBS::new_from(&orig, Some(serialized_context.clone()));
```
Comment (franziskuskiefer) [inline src/framing/test_framing.rs:82]: This only exercises the `Some` case. We should also have a test for the `None` case.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
One more 🙂

File: src/framing/test_framing.rs (hunk 2 above)
Comment (raphaelrobert) [reply src/framing/test_framing.rs:60]: Good point, I filed #121

**Review by franziskuskiefer — APPROVED** [review]
thanks
